### PR TITLE
Fix draft order tax mismatch

### PR DESCRIPTION
**Pull request by IKarbowiak**
Tax for draft order calculated by us and returned by Avalara differ. It's caused because of division total price be quantity and then multiplying the unit price by this quantity.

This PR introduces `calculate_order_line_total` plugin method that returns the total price for the order, returned by the plugin.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-2715 Avalara draft orders tax mismatch](https://app.clickup.com/t/2549495/SALEOR-2715)

**Review by tomaszszymanski129 — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

File: saleor/plugins/avatax/plugin.py [hunk 1]
```diff
@@ -305,19 +305,50 @@ def calculate_checkout_line_total(
...
+            return previous_value
+
+        if not _validate_order(order):
+            return zero_taxed_money(order.total.currency)
+
+        taxes_data = self._get_order_tax_data(order, previous_value)
+        return self._calculate_total_price(taxes_data, variant.sku, previous_value)
+
+    @staticmethod
+    def _calculate_total_price(
```
Comment (fowczarek) [inline saleor/plugins/avatax/plugin.py:333]: Maybe `_calculate_line_total_price`

File: saleor/plugins/avatax/tests/test_avatax.py [hunk 2]
```diff
@@ -97,6 +97,97 @@ def test_calculate_checkout_line_total(
...
+        order_line.order,
+        order_line,
+        variant,
+        product,
+    )
+    total = quantize_price(total, total.currency)
+    assert total == TaxedMoney(net=Money("30.00", "USD"), gross=Money("36.90", "USD"))
+
+
+@pytest.mark.vcr()
```
Comment (fowczarek) [inline saleor/plugins/avatax/tests/test_avatax.py:149]: Do we need a VCR here? Currently, we don't have any cassettes for this test.

**Review by fowczarek — APPROVED** [review]

File: saleor/plugins/avatax/plugin.py [hunk 3]
```diff
@@ -305,19 +305,50 @@ def calculate_checkout_line_total(
...
+        order: "Order",
+        order_line: "OrderLine",
+        variant: "ProductVariant",
+        product: "Product",
+        previous_value: TaxedMoney,
+    ) -> TaxedMoney:
+        if self._skip_plugin(previous_value):
+            return previous_value
+
+        if not variant or (variant and not product.charge_taxes):
```
Comment (korycins) [inline saleor/plugins/avatax/plugin.py:323]: The typing for calculate_order_line_total doesn't assume that you will receive Optional variant here.

Comment (korycins) [reply saleor/plugins/avatax/plugin.py:323]: Also, I would say that `if not variant or not product.charge_taxes:` should be enough here 🤔

File: saleor/order/utils.py [hunk 4]
```diff
@@ -193,12 +193,13 @@ def update_taxes_for_order_line(
     line_price = line.unit_price.gross if tax_included else line.unit_price.net
     line.unit_price = TaxedMoney(line_price, line_price)
 
-    price = manager.calculate_order_line_unit(order, line, variant, product)
```
Comment (korycins) [inline saleor/order/utils.py:196]: What about second place where we call `manager.calculate_order_line_unit` ?
`add_variant_to_order` has logic
```
        unit_price = manager.calculate_order_line_unit(order, line, variant, product)
        line.unit_price = unit_price
        line.total_price = unit_price * quantity
        line.tax_rate = manager.get_order_line_tax_rate(
            order, product, variant, None, unit_price
        )
        line.total_price = unit_price * quantity
```

**Review by korycins — APPROVED** [review]

**Pull request merged**